sdk/program/src/lib.rs

```diff
@@ -1,3 +1,650 @@
+//!   program.
+//!   - ID: [`solana_sdk::compute_budget`](https://docs.rs/solana-sdk/latest/solana_sdk/compute_budget/index.html)
+//!   - Instruction: [`solana_sdk::compute_budget`](https://docs.rs/solana-sdk/latest/solana_sdk/compute_budget/index.html)
+//!
+//! ## Precompiles
+//!
+//! - __ed25519 Program__: Verifies an ed25519 signature.
+//!   - ID: [`solana_program::ed25519_program`]
+//!   - Instruction: [`solana_sdk::ed25519_instruction`](https://docs.rs/solana-sdk/latest/solana_sdk/ed25519_instruction/index.html)
+//!
+//! - __secp256k1 Program__: Verifies secp256k1 public key recovery operations.
+//!   - ID: [`solana_program::secp256k1_program`]
+//!   - Instruction: [`solana_sdk::secp256k1_instruction`](https://docs.rs/solana-sdk/latest/solana_sdk/secp256k1_instruction/index.html)
+//!
+//! ## Program loaders
+//!
+//! - __BPF Loader__: Deploys, and executes immutable programs on the chain.
+//!   - ID: [`solana_program::bpf_loader`]
+//!   - Instruction: [`solana_program::loader_instruction`]
+//!
+//! - __Upgradable BPF Loader__: Deploys, upgrades, and executes upgradable
+//!   programs on the chain.
+//!   - ID: [`solana_program::bpf_loader_upgradeable`]
+//!   - Instruction: [`solana_program::loader_upgradeable_instruction`]
+//!
+//! - __Deprecated BPF Loader__: Deploys, and executes immutable programs on the
+//!   chain.
+//!   - ID: [`solana_program::bpf_loader_deprecated`]
+//!   - Instruction: [`solana_program::loader_instruction`]
+//!
+//! # Sysvars
+//!
+//! Sysvars are special accounts that contain dynamically updating data about
+//! the network cluster, the blockchain history, and the executing transaction.
+//!
+//! The program IDs and deserializers for sysvars are defined in the [`sysvar`]
+//! module, and simple sysvars implement the [`Sysvar`] trait. Since to a Solana
+//! program sysvars are just accounts, if the `AccountInfo` is provided to the
+//! program, then the program can deserialize the sysvar with
+//! [`Sysvar::from_account_info`] to access its data, as in this example that
+//! logs the [`clock`][clk] sysvar.
+//!
+//! [`Sysvar`]: sysvar::Sysvar
+//! [`Sysvar::from_account_info`]: sysvar::Sysvar::from_account_info
+//! [clk]: sysvar::clock
+//!
+//! ```
+//! use solana_program::{
+//!     account_info::{next_account_info, AccountInfo},
+//!     clock,
+//!     entrypoint::ProgramResult,
+//!     msg,
+//!     pubkey::Pubkey,
+//!     sysvar::Sysvar,
+//! };
+//!
+//! fn process_instruction(
+//!     program_id: &Pubkey,
+//!     accounts: &[AccountInfo],
+//!     instruction_data: &[u8],
+//! ) -> ProgramResult {
+//!     let account_info_iter = &mut accounts.iter();
+//!     let clock_account = next_account_info(account_info_iter)?;
+//!     let clock = clock::Clock::from_account_info(&clock_account)?;
+//!     msg!("clock: {:#?}", clock);
+//!     Ok(())
+//! }
+//! ```
+//!
+//! As an optimization, to avoid including sysvar accounts in instructions and deserializing them
```

Comment:
This is the preferred way and should be highlighted first.  The account method of getting sysvars should be clearly described as a way to deal older programs that still pass sysvars as accounts.